### DISCUSSION TITLE
What about completely new text lines?

### DISCUSSION THREAD

**Discussion by PeterSheeper** [humhue/DAOTools#9] [Q&A] [answered]
Hello, humhue,
Firstly Thank you for the tool and your work on it. QUDAO fixpack from your “Releases” works very well on the Polish localisation of DA:O.
I just have a question concerning how your tool manages text lines in modded dlg files that are not present in “singleplayer_en-us.tlk” and therefore any other localised version of this file. I am talking about completely new text lines added by a mod.
I don’t fully understand if the tool keeps string data or replacing the string offset to 0 masks local variables altogether because when I open dlg files edited by your tool in pyGFF local strings don't show up. Do you happen to know if those text lines without a reference in localised xml files will show printing local strings from dlg file (so in the case of QUDAO: in English) or the game might go crazy😅?
Unfortunately, I cannot tell what exactly happens since those text lines are in quite specific places in the game and I just haven’t encountered them yet. Theoretically, I suppose that the game will just skip them. Do you think that adding those references manually to “singleplayer_en-us.tlk” or in my case “singleplayer_pl-pl.tlk” might solve the problem?

**Comment by humhue** [comment, marked as answer]
Hey, I can't understand what do you want to know, what you're trying to do and why you're trying to do it, like, what's even the problem?

However, you can find a little insight here, about how the patcher works:
https://github.com/humhue/DAOTools#how-does-it-work

**Reply by PeterSheeper** [reply]
You are right. It's not a problem. It's just that for some lines in some mods text strings are only inside dlg files and your program deletes those text strings. The game doesn't have any text to show. But on the other hand, it's obvious, so I don't know why I asked that. Never mind, That's not too difficult to fix so I will manage. Sorry for the unnecessary trouble. You can delete this discussion not to create a mess on your repository

**Reply by humhue** [reply]
This is a perfectly reasonable thing to point out, and you indeed got a point.
The patcher will eventually strip the references out of some new texts, if any, in .dlg, .cut and .plo files packaged inside .erf files. Thing is, usually a modder will put these very texts in a specific .tlk file, or maybe inside an .erf file of their own.
I tried looking for the maximum basegame TLK string id, and I couldn't find it, maybe I didn't search it enough, but, by knowing it, the patcher could discriminate old and new (not to patch) texts.
Thanks for reaching out (and for the star lul)

**Reply by PeterSheeper** [reply]
Ok, I hear you. Even though it looks like new lines in mods made in the Toolset have ids greater than 610000000 (but that is just my unconfirmed observation), if a dlg was edited in pyGff a modder could set string Id to be anything including those values  that are already taken so it's rather impossible to detect a new string line (besides searching all tlk files for each parsed line but I guess that's not the solution xd). I didn't know about that .tlk part. All is clear to me now. I think you've given me an idea. Thanks a lot 👍

**Reply by humhue** [reply]
I guess I'll check if that's really the starting id for the new lines, if it is, I might skip the values whose id is greater than that. It would be a good idea.

**Reply by humhue** [reply]
Hey @PeterSheeper, I know it's been a long time. The program should now let you manually translate the tlk strings outside the normal range (those added by modders). And also comes with a GUI. Thanks for this discussion :)

**Reply by PeterSheeper** [reply]
Hi, Thanks for the update. I'm glad the tool supports this now. Congrats on the v2.0.0 release!
